R/fwrite.R

```diff
@@ -51,6 +51,17 @@ fwrite <- function(x, file="", append=FALSE, quote="auto",
     showProgress = FALSE
     eol = "\n"  # Rprintf() is used at C level which knows inside it to output \r\n on Windows. Otherwise extra \r is output.
   }
+  if (ncol(x) == 0L && file != "") {
+    # should cover append = TRUE case as well
+    if (file.exists(file)) {
+      warning("Input has no columns; doing nothing.")
```

Comment:
what if file exists and is not empty and we used `append=FALSE`, file should be overwritten then, isn't it? otherwise there is small incosistency

Reply:
Agree it can be painful, so if we sacrifice consistency, could we mention that in warning? something like
```r
"Input has no columns"
if(!append & filesize>0) "empty file has not been materialized to not overwrite existing non-empty file"
"doing nothing."

```